cloudinit/ssh_util.py

```diff
@@ -211,32 +213,58 @@ def users_ssh_info(username):
     return (os.path.join(pw_ent.pw_dir, '.ssh'), pw_ent)
 
 
-def extract_authorized_keys(username):
+def render_authorizedkeysfile(value, homedir, username):
+    if value is None:
+        value = "%h/.ssh/authorized_keys"
+    for macro, field in (("%h", homedir), ("%u", username), ("%%", "%")):
+        value = value.replace(macro, field)
+        if not value.startswith("/"):
+            value = os.path.join(homedir, value)
+    return value
+
+
+def extract_authorized_keys(username, sshd_cfg_file=DEF_SSHD_CFG):
     (ssh_dir, pw_ent) = users_ssh_info(username)
-    auth_key_fn = None
+    auth_key_fns = []
     with util.SeLinuxGuard(ssh_dir, recursive=True):
```

Comment:
I dont think we need this... we only read the file here.
so I dont think this does anything.

Reply:
Agreed